server/events_test.go

```diff
@@ -1107,3 +1107,40 @@ func TestServerEventStatsZ(t *testing.T) {
 		t.Fatalf("Expected a route, but got %d", lr)
 	}
 }
+
+func TestServerEventsPingStatsZ(t *testing.T) {
+	sa, _, sb, optsB, akp := runTrustedCluster(t)
+	defer sa.Shutdown()
+	defer sb.Shutdown()
+
+	url := fmt.Sprintf("nats://%s:%d", optsB.Host, optsB.Port)
+	nc, err := nats.Connect(url, createUserCreds(t, sb, akp))
+	if err != nil {
+		t.Fatalf("Error on connect: %v", err)
+	}
+	defer nc.Close()
+
+	reply := nc.NewRespInbox()
+	sub, _ := nc.SubscribeSync(reply)
+
+	nc.PublishRequest(serverStatsPingReqSubj, reply, nil)
+
+	// Make sure its a statsz
+	m := ServerStatsMsg{}
+
+	// Receive both manually.
+	msg, err := sub.NextMsg(randomBackoff)
```

Comment:
I would recommend that you add some time padding here to avoid flapping. Since the server may sleep up to that amount of time and then send the request, I see that NextMsg() call could be timing out in some builds on Travis.

Reply:
No, but just seeing all those flapping tests these days on Travis, I would rather have an extra wait in case, than have to recycle the whole test suite.